internal/provider/cluster_resource_test.go

```diff
@@ -975,6 +982,17 @@ func TestIntegrationServerlessClusterResource(t *testing.T) {
 			},
 			finalCluster: singleRegionClusterWithUnlimited("BASIC"),
 		},
+		{
+			name: "upgrade cockroach_version for STANDARD cluster",
```

Comment:
This is the only real automated test for a successful standard tier upgrade.  Our api doesn't allow setting up these tests since we need to be able to build a host cluster and then upgrade it.  For that reason, we're currently limited to integration tests here.   Given the complication of upgrades it would be great to figure out a way to build an acceptance test for this.  @andy-kimball Do you have any ideas for that?

Reply:
@alyshanjahani-crl, do we support creating a Serverless cluster at the previous version? If so, we could do that, and then upgrade it to the next version, all without touching the host cluster version.

Reply:
We're currently limited creating serverless clusters at the latest version. This is why cockroach_version is not accepted for serverless clusters during the Create phase.  But some kind of ability in the future would be welcome.  We could hide it behind a feature flag if need be.